lib/ocpp/common/websocket/websocket_tls.cpp

```diff
@@ -14,6 +14,41 @@
 
 namespace ocpp {
 
+// verify that the csms certificate's commonName matches the CSMS FQDN
+bool verify_csms_cn(const std::string& hostname, bool preverified, boost::asio::ssl::verify_context& ctx) {
+
+    if (!preverified) {
+        EVLOG_error << "Could not verify CSMS server certificate";
+        return false;
+    }
+
+    int depth = X509_STORE_CTX_get_error_depth(ctx.native_handle());
+
+    // only check for CSMS server certificate
+    if (depth == 0) {
+        // Get server certificate
+        X509* server_cert = X509_STORE_CTX_get_current_cert(ctx.native_handle());
```

Comment:
We should probably use the types from libevse-security here https://github.com/EVerest/libevse-security/blob/main/include/evse_security/detail/openssl/openssl_types.hpp

Reply:
How about moving this functionality as a neat helper function to `libevse-security` instead? 

Then all the OpenSSL-specifics here aren't part of `libocpp`. I mean in the midterm, that's the plan with `libevse-security`, right?